R/splitBy.R

```diff
@@ -109,16 +109,26 @@ setGeneric("splitBy",
     }
     # Check f or extract the factor from rowData or colData
     if( !.is_non_empty_string(f) ){
-        if(length(f) > 1L && (is.character(f) || is.numeric(f))){
+        if( length(f) > 1L ){
             f <- factor(f, unique(f))
         }
-        if(length(f) %in% dim(x)){
+        # Check if the length of f matches with one of the dimensions
+        if(!length(f) %in% dim(x)){
             stop("'f' must either be a single non-empty character value or",
                  " vector coercible to factor alongside one of the ",
                  "dimensions of 'x'",
                  call. = FALSE)
         }
+        # Get the dimension that matches
         MARGIN <- which(dim(x) %in% length(f))
+        # If it matches with both directions
+        if(length(MARGIN) > 1 ){
+            # Get 2 if f can be found from colData, otherwise get 1
+            MARGIN <- ifelse(any( sapply(colData(x), 
+                                         function(var){all.equal(as.character(var), 
+                                                                 as.character(f))}) == TRUE ), 
+                             2, 1)
+        }
```

Comment:
A bit too complicated

Reply:
Or if MARGIN could be specified by user (hidden option?)

Reply:
From my point of view a solution to that problem is available. You have outlined in the example above. So from my point of view, it falls in the category of the user should know. Otherwise it will only get more complicated on our part without adding any benefit for the user.